src/NHibernate.Test/Linq/TryGetMappedTests.cs

```diff
@@ -807,7 +819,7 @@ public void NotNhQueryableTest()
 			Assert.That(() => expectedMemberType(memberType), $"Invalid member type: {memberType?.Name ?? "null"}");
 			Assert.That(() => expectedComponentType(componentType), $"Invalid component type: {componentType?.Name ?? "null"}");
 
-			if (found)
+			if (found && (componentType == null || componentType.PropertyNullability != null))
```

Comment:
Maybe instead make nullability parameter nullable. And check that `_tryGetMappedNullability` returns false when it's null.

Reply:
I adding a commit doing what I think you are meaning, but I do not see this as an improvement. It looks to me as a less understandable way of testing, and maybe more brittle, subject to cause the test to fail if we start implementing that nullability property for custom composite type.

Reply:
But with your initial change `TryGetMappedNullability` call is not tested at all for problematic case. Isn't it?  So your changes were not covered.

> subject to cause the test to fail if we start implementing that nullability property for custom composite type.

I see no problem with adjusting existing test cases when behavior is changed.

nit: I would also add optional `nullability` parameter to `AssertResult` method and keep `AssertResult` call it test case

Reply:
Ok, I thought the test parsing the query would also cause the method to be called, but I had not checked. I have added your nit.